src/main/java/com/limechain/client/FullNode.java

```diff
@@ -69,7 +71,13 @@ public void start() {
 
         // Start syncing
         log.log(Level.INFO, "Node successfully connected to a peer! Sync can start!");
-        AppBean.getBean(WarpSyncMachine.class).start();
+
+        CliArguments args = AppBean.getBean(CliArguments.class);
+        if ("full".equals(args.syncMode().toLowerCase())) {
+            AppBean.getBean(FullSyncMachine.class).start();
+        } else {
+            AppBean.getBean(WarpSyncMachine.class).start();
+        }
```

Comment:
(no action required) 

1. Maybe use a switch statement
2. I believe it'd be better to have some validation whether the sync mode is a valid string instead of wating to find peers.

Reply:
Currently it fallbacks to warp if the input is invalid, but yes it would be troublesome if someone just mistyped full and it fallbacks to warp